### PR TITLE
Julia Programming Documentation Updates

### PR DESCRIPTION
**Pull request by wilsonhj**
- Changed all absolute links in /docs/src to relative links (@ref)
- Changed all relative file paths to relative links (@ref)
-DID NOT change any relative paths that point to a .png, i.e 
_Better Automatic Solution_
because I am not yet sure whether it is necessary to include the file extension .png, i.e
 _Better Automatic Solution_

Hope this helps!

### CODE REVIEW

File: docs/src/analysis/bifurcation.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Bifurcation analysis on DifferentialEquations.jl types can be performed by:
 - PyDSTool.jl
 - DiffEqBiological.jl
 
-If your system is a chemical reaction system, see the documentation at 
-[DiffEqBiological.jl](https://github.com/JuliaDiffEq/DiffEqBiological.jl#making-bifurcation-diagram)
+If your system is a chemical reaction system, see the documentation at
+[DiffEqBiological.jl](@ref)
```
Comment (ChrisRackauckas) [inline docs/src/analysis/bifurcation.md]: this is not correct: `@ref` doesn't work for external links.

File: docs/src/types/rode_types.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ page.
 * `p`: The optional parameters for the problem. Defaults to `NullParameters`.
 * `noise`: The noise process applied to the noise upon generation. Defaults to
   Gaussian white noise. For information on defining different noise processes,
-  see [the noise process documentation page](../../../features/noise_process)
+  see [the noise process documentation page](@ref noise_process)
```
Comment (ChrisRackauckas) [inline docs/src/types/rode_types.md]: This is only correct if a label is put on the page, or it matches the title exactly.

**Comment by ChrisRackauckas** [comment]
The vast majority of these changes are incorrect. The two repeated issues are found in the comments above. I would recommend trying to build locally to check the resulting link structure.

**Comment by wilsonhj** [comment]
Ok. I kind of figured. I’ll have to do this again. How are you building
locally? Are you running Julia 1.x in Atom? Is there documentation for @ref
@id ?

**Comment by ChrisRackauckas** [comment]
>How are you building locally? 

https://github.com/JuliaDiffEq/DiffEqDocs.jl/blob/master/docs/make.jl#L1-L112

>Is there documentation for @ref @id ?

https://juliadocs.github.io/Documenter.jl/stable/man/syntax/#@ref-link-1 and search around for id.

**Comment by wilsonhj** [comment]
Chris, I want to do this correctly. Links that can be replaced with @ref
must be in the juliaDiffEq codebase? The headers must match exactly to be
replaced by @ref?

**Comment by ChrisRackauckas** [comment]
>Chris, I want to do this correctly. Links that can be replaced with @ref must be in the juliaDiffEq codebase? The headers must match exactly to be replaced by @ref?

It must match the header or an id placed on it.

@kanav99 help explain if you can. But I really think

https://github.com/JuliaDiffEq/DiffEqDocs.jl/pull/309/files

is sufficient documentation of the idea.

**Comment by kanav99** [comment]
Hey @wilsonhj The issue you are trying to fix is something we already are trying to fix (#309). You should refer to [Julia Documentation](https://docs.julialang.org/en/v1/manual/documentation/) page (for seeing how docstrings work and [Documenter.jl](https://juliadocs.github.io/Documenter.jl/stable/) to see how you deploy it to a webpage.
Broken links issue is already fixed by #309. So you can probably help us find and fix any further issues if you feel so.

**Comment by ChrisRackauckas** [comment]
Instead of focusing on links, @wilsonhj how about grammar/typos for a page?

**Pull request closed without merging**